### PR TITLE
FMReader: remove manga title from chaper names

### LINKED ISSUE #3420: [Feature Request] FMReader: remove manga titles from chapter names
### What feature should be added to this source?
with this manga from LHT: https://lhtranslation.net/manga-shikkaku-mon-no-saikyou-kenja.html
one chapter name looks like this: 
`SHIKKAKU MON NO SAIKYOU KENJA - SEKAI SAIKYOU NO KENJA GA SARA NI TSUYOKUNARU TAME NI TENSEI SHIMASHITA Chapter 34`
the only real useful information is `Chapter 34`, but it's out of reach because the chapter name is so damn long that even in landscape mode you couldnt read it
this redundancy should be removed.

### PR DESCRIPTION
**Pull request by AriaMoradi**
closes #3420

### CODE REVIEW

**Comment by SnakeDoc83** [comment]
Update build.gradle, remove log.

File: src/all/fmreader/src/eu/kanade/tachiyomi/extension/all/fmreader/FMReader.kt [hunk 1]
```diff
@@ -101,7 +102,8 @@ abstract class FMReader(
         val mangas = document.select(popularMangaSelector()).map { popularMangaFromElement(it) }
 
         // check if there's a next page
-        val hasNextPage = (document.select(popularMangaNextPageSelector())?.first()?.text() ?: "").let {
+        val hasNextPage = (document.select(popularMangaNextPageSelector())?.first()?.text()
```
Comment (SnakeDoc83) [inline src/all/fmreader/src/eu/kanade/tachiyomi/extension/all/fmreader/FMReader.kt:105]: Is this just a newline that's been added?

Comment (AriaMoradi) [reply src/all/fmreader/src/eu/kanade/tachiyomi/extension/all/fmreader/FMReader.kt:105]: looks like reformating the file changed it

**Comment by AriaMoradi** [comment]
changes done

**Comment by SnakeDoc83** [comment]
Just tested this--it's going to need more work if you want this committed. As it is it breaks some of the sources.

**Comment by AriaMoradi** [comment]
@SnakeDoc83 
I didn't understand how it breaks sources?
the only change is: if chapter name is present remove it, if not dont do anything
would you present a few examples for me to investigate?

**Comment by AriaMoradi** [comment]
also I did a test:
first I downloaded a chapter with the current version of the code, then installed my patched one and the downloaded chapter was gone, so
1- if a source changes a chapter's name but not link, tachiyomi loses the data about it, so maybe this is a bug and should be fixed inside the app (only use the link as unique data?)
2- the same problem would exist with the same patch with my patch to `Paprika`

**Comment by SnakeDoc83** [comment]
Take a look at FMReaderFactory.kt and you'll see some sources were overriding or relying on `chapterFromElement` as it was. I don't have Android Studio open at the moment, but as I recall at least 2 of the non-English sources wouldn't display chapter lists.

**Comment by AriaMoradi** [comment]
> Take a look at FMReaderFactory.kt and you'll see some sources were overriding or relying on `chapterFromElement` as it was. I don't have Android Studio open at the moment, but as I recall at least 2 of the non-English sources wouldn't display chapter lists.

fixed

File: src/all/fmreader/src/eu/kanade/tachiyomi/extension/all/fmreader/FMReader.kt (hunk 1 above)
Comment (SnakeDoc83) [reply src/all/fmreader/src/eu/kanade/tachiyomi/extension/all/fmreader/FMReader.kt:104]: It's a nitpick but I'd prefer you return it to how it was for readability. Also remove the one commented out log.d and then I'll merge.

Comment (AriaMoradi) [reply src/all/fmreader/src/eu/kanade/tachiyomi/extension/all/fmreader/FMReader.kt]: done and done

**Pull request merged**